src/wp-includes/blocks/template-part.php

```diff
@@ -275,8 +275,8 @@ function register_block_core_template_part() {
 	register_block_type_from_metadata(
 		__DIR__ . '/template-part',
 		array(
-			'render_callback' => 'render_block_core_template_part',
-			'variations'      => build_template_part_block_variations(),
+			'render_callback'    => 'render_block_core_template_part',
+			'variation_callback' => 'build_template_part_block_variations',
```

Comment:
We are going to need some conditional logic here, as these PHP is stored in Gutenberg that supports two versions of WP back.

Reply:
I'm not sure we need conditional logic here, because this change will only be made in versions of GB or Core that support the `variation_callback` registration arg. What's more important is ensuring that previous versions of GB which build these variations during registration still work as expected once the changes to `WP_Block_Type` are merged in core. I believe our tests will already account for this, since when this is merged we'll still have the previous block definitions in place.